txtdirect.go

```diff
@@ -73,31 +73,37 @@ func contains(array []string, word string) bool {
 // and if it's not provided it will check txtdirect config for
 // default fallback address
 func fallback(w http.ResponseWriter, r *http.Request, fallback, recordType, fallbackType string, code int, c Config) {
-	if fallback != "" {
-		log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, fallback)
-		if code == http.StatusMovedPermanently {
-			w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
-		}
-		w.Header().Add("Status-Code", strconv.Itoa(code))
+	if code == http.StatusMovedPermanently {
+		w.Header().Add("Cache-Control", fmt.Sprintf("max-age=%d", status301CacheAge))
+	}
+	w.Header().Add("Status-Code", strconv.Itoa(code))
+
+	if fallback != "" && fallbackType != "global" {
 		http.Redirect(w, r, fallback, code)
 		if c.Prometheus.Enable {
 			FallbacksCount.WithLabelValues(r.Host, recordType, fallbackType).Add(1)
 			RequestsByStatus.WithLabelValues(r.URL.Host, strconv.Itoa(code)).Add(1)
 		}
+	} else if contains(c.Enable, "www") {
+		s := strings.Join([]string{defaultProtocol, "://", defaultSub, ".", r.URL.Host}, "")
+		if c.Prometheus.Enable {
+			RequestsByStatus.WithLabelValues(r.URL.Host, strconv.Itoa(code)).Add(1)
+		}
+		http.Redirect(w, r, s, code)
 	} else if c.Redirect != "" {
-		if contains(c.Enable, "www") {
-			log.Printf("[txtdirect]: %s > %s", r.Host+r.URL.Path, c.Redirect)
-			w.Header().Set("Content-Type", "")
-			w.Header().Add("Status-Code", strconv.Itoa(http.StatusForbidden))
-			http.Redirect(w, r, c.Redirect, http.StatusForbidden)
-			if c.Prometheus.Enable {
-				FallbacksCount.WithLabelValues(r.Host, recordType, "global").Add(1)
-				RequestsByStatus.WithLabelValues(r.URL.Host, string(http.StatusForbidden)).Add(1)
-			}
+		w.Header().Set("Content-Type", "")
+		w.Header().Set("Status-Code", strconv.Itoa(http.StatusForbidden))
+
+		http.Redirect(w, r, c.Redirect, http.StatusForbidden)
+
+		if c.Prometheus.Enable {
+			FallbacksCount.WithLabelValues(r.Host, recordType, "global").Add(1)
```

Comment:
Using "global" ass fallbackType within the function, which includes both 404, "redirect" and "www" seems good, but I think we can split this up within the metrics:
404 -> no fallback count
redirect used -> "redirect"
www used -> "subdomain"

Or similar